File: ios/Classes/models/AutoSelfieSettings.swift
//
//  AutoSelfieSettings.swift
//  flutter_amanisdk_v2
//
//  Created by Deniz Can on 14.11.2022.
//

import Foundation


struct AutoSelfieSettings: Codable {
  // info messages
  var faceIsOk: String = "Lütfen sabit durun"
  var notInArea: String = "Yüzünüzü daire içine hizalayın"
  var faceTooSmall: String = "Yüzünüz çok uzakta"
  var faceTooBig: String = "Yüzünüz çok yakında"
  var completed: String = "İşlem Tamam"
  // screen config
  var appBackgroundColor: String = "000000"
  var appFontColor: String = "ffffff"
  var primaryButtonBackgroundColor: String = "ffffff"
  var ovalBorderSuccessColor: String = "00ff00"
  var ovalBorderColor: String = "ffffff"
  var countTimer: String = "3"
  var manualCropTimeout: Int = 30
}

File: lib/common/models/ios/auto_selfie_settings.dart
class IOSAutoSelfieSettings {
  // info messages
  String? faceIsOk;
  String? notInArea;
  String? faceTooSmall;
  String? faceTooBig;
  String? completed;
  // screen config
  String? appBackgroundColor;
  String? appFontColor;
  String? primaryButtonBackgroundColor;
  String? ovalBorderSuccessColor;
  String? ovalBorderColor;
  String? countTimer;
  int? manualCropTimeout;

  IOSAutoSelfieSettings(
      {this.faceIsOk,
      this.notInArea,
      this.faceTooSmall,
      this.faceTooBig,
      this.completed,
      this.appBackgroundColor,
      this.appFontColor,
      this.primaryButtonBackgroundColor,
      this.ovalBorderColor,
      this.ovalBorderSuccessColor,
      this.countTimer,
      this.manualCropTimeout});

  Map toJson() {
    return {
      "faceIsOk": faceIsOk,
      "notInArea": notInArea,
      "faceTooSmall": faceTooSmall,
      "faceTooBig": faceTooBig,
      "completed": completed,
      "appBackgroundColor": appBackgroundColor,
      "appFontColor": appFontColor,
      "primaryButtonBackgroundColor": primaryButtonBackgroundColor,
      "ovalBorderSuccessColor": ovalBorderSuccessColor,
      "ovalBorderColor": ovalBorderColor,
      "countTimer": countTimer,
      "manualCropTimeout": manualCropTimeout,
    };
  }
}
